### DISCUSSION TITLE
Use "Files" instead of "FileSystem" in the UI

### DISCUSSION THREAD

**Discussion by andyprice** [godotengine/godot-proposals#9067] [Editor]
"FileSystem" is a fine name for a C++ class but it's not a normal word and looks inconsistent with the rest of Godot's UI labels. I think it would be better to use "Files" or maybe "File Tree" in the UI. "Files" would be my preference as it's simple and probably minimizes translation issues.

Current:
![Screenshot from 2024-02-10 16-21-49](https://github.com/godotengine/godot-proposals/assets/133737/8e45e3ab-24ef-404a-b9ef-4cb946abe5a8)

Proposed:

https://github.com/godotengine/godot-proposals/assets/133737/ad9677a8-c822-4564-83d1-957c581a4b52

[Edit: Added a better screenshot and a video to show the proposal better]

**Comment by AThousandShips** [comment]
This wasn't intended to be visible I believe, it is just the automatic name picked when moving it to the bottom because of how that works

**Reply by andyprice** [reply]
This is unrelated to the change that allowed moving the FileSystem pane to the bottom. Perhaps I should have made my screenshots with it in the default position to avoid confusion. Godot has been using "FileSystem" in the UI for as long as I've been using it.

**Reply by AThousandShips** [reply]
It's the same cause, but yes that would have been clearer, the name is added and not translated

**Reply by andyprice** [reply]
I imagine the essence of the change I'm proposing would look like:

```diff
diff --git a/editor/editor_node.cpp b/editor/editor_node.cpp
index 07a048c607..8443d4936e 100644
--- a/editor/editor_node.cpp
+++ b/editor/editor_node.cpp
@@ -7068,7 +7068,7 @@ EditorNode::EditorNode() {
        editor_dock_manager->add_control_to_dock(EditorDockManager::DOCK_SLOT_LEFT_UR, ImportDock::get_singleton(), TTR("Import"));
 
        // FileSystem: Bottom left.
-       editor_dock_manager->add_control_to_dock(EditorDockManager::DOCK_SLOT_LEFT_BR, FileSystemDock::get_singleton(), TTR("FileSystem"));
+       editor_dock_manager->add_control_to_dock(EditorDockManager::DOCK_SLOT_LEFT_BR, FileSystemDock::get_singleton(), TTR("Files"));
 
        // Inspector: Full height right.
        editor_dock_manager->add_control_to_dock(EditorDockManager::DOCK_SLOT_RIGHT_UL, InspectorDock::get_singleton(), TTR("Inspector"));
@@ -7089,7 +7089,7 @@ EditorNode::EditorNode() {
        default_layout.instantiate();
        // Dock numbers are based on DockSlot enum value + 1.
        default_layout->set_value(docks_section, "dock_3", "Scene,Import");
-       default_layout->set_value(docks_section, "dock_4", "FileSystem");
+       default_layout->set_value(docks_section, "dock_4", "Files");
        default_layout->set_value(docks_section, "dock_5", "Inspector,Node,History");
 
        // There are 4 vsplits and 4 hsplits.
diff --git a/editor/filesystem_dock.cpp b/editor/filesystem_dock.cpp
index 02ae15b80d..046512a7de 100644
--- a/editor/filesystem_dock.cpp
+++ b/editor/filesystem_dock.cpp
@@ -3740,7 +3740,7 @@ void FileSystemDock::load_layout_from_config(Ref<ConfigFile> p_layout, const Str
 
 FileSystemDock::FileSystemDock() {
        singleton = this;
-       set_name("FileSystem");
+       set_name("Files");
        current_path = "res://";
 
        ProjectSettings::get_singleton()->add_hidden_prefix("file_customization/");
```

**Reply by AThousandShips** [reply]
This isn't sufficient, it won't work with the bottom dock, it is done via a way that doesn't translate the string

**Reply by Calinou** [reply]
The current docs and third-party resources extensively refer to the FileSystem dock, so renaming it to a Files dock could be confusing for some. I'm not strictly opposed to the idea, but remember we have no control over third-party resources.

**Reply by andyprice** [reply]
That's a good point. Perhaps we can revisit the idea for the next major version.

**Reply by passivestar** [reply]
An argument could be made that for every release that still calls it FileSystem more third-party resources will be created that refer to it by that name. Godot tutorial space is still young, maybe it's a good time to do those kind of changes 🙂

**Reply by AThousandShips** [reply]
I'd say that's not a reason enough to change something that has little practical benefit, it's not critical, so the confusion is more of an issue than the change itself, if this was a critical issue then that'd be a different question, but I'd say that it'd be a bigger problem to have inconsistencies here than having a sub-optimal name

**Reply by passivestar** [reply]
Is the plan to batch a bunch of these changes into a version that will effectively be a **blender 2.8**, where everything changes? I hope we're not talking about godot 5.0 here 😅

**Reply by AThousandShips** [reply]
Well I don't see how it's any better to confuse people with a new name and create inconsistencies with third party content etc. than to have a cumbersome but well established name, if it's not something that we can say is worth that confusion we shouldn't do it until we break that kind of compatibility, remember that this name was used in 2.x and 3.x as well so it'd cause possible confusion for migration as well, it's well established

**Reply by passivestar** [reply]
yeah I understand that, I'm just wondering how Godot handles introducing these kinds of changes, because I don't know. From what I understand there's no established annual release cycle for major versions or anything like that. And it's not all about fixing critical issues either, I'm sure there are a lot of proposed changes that break compatibility but still necessary, right? How often do those happen in Godot?

**Reply by AThousandShips** [reply]
If they're critical enough in breaking change we don't do them until the next major version, if we do them at all, when we can't guarantee stability or effectiveness of that change

If it's critical we accept compatibility breaking in different cases, but if the feature works but not completely we avoid compatibility breakage as much as possible, at least if there's no option to migrate it, something can break compatibility while also having methods of backwards compatibility, when those methods aren't seamless or invisible, we've changed a number of things during 4.x that broke compatibility but we avoid it as much as possible

For things that are just aesthetic or limitations, we hold off from it because it's just not worth it in some cases

This might be open for this, my personal preference here would be to just change it to "File System" to make it less clunky, but it's not critical in either direction IMO so it's something I'm open to either direction with, I don't necessarily see that changing it would be impossible, but I also don't think changing it at all is necessarily needed either, in 4.x or 5.x

Also I'd say that "Files" is *less* obvious and harder to both identify *and* translate, so I'd say naming it "File System" is the best way, it doesn't disrupt anything really, and it doesn't make it ambiguous, "Files" could be many things, like "The files in this scene" or "the files of the project", it isn't as clear IMO (would also need to be contextual so it doesn't translate *other* things in the editor named "Files")

**Reply by passivestar** [reply]
thanks for the explanation!

**Reply by RPicster** [reply]
For me, I don't care too much if it's called "Files", "File Browser", "File Explorer" or something else. I just join this discussion as I don't think changing what is written on a Label could be considered a "breaking change". What exactly would break in the engine by changing what is written on the label?

External resources or tutorials are outdated all the time - just think of all the particle tutorials that are deprecated because of the redesign which happened with a minor version change. Stuff is changed all the time and from my personal point of view: It's ok! Users know that Godot is in development.

It feels a bit like shutting down this proposal or the room for such discussions and I don't think the Godot Editor UI/UX is in a state where this is a good approach.

**Reply by AThousandShips** [reply]
No one is shutting down anything, how is an active discussion "shut down"?

**Reply by RPicster** [reply]
From my perspective it looks like the value of this proposal is given less importance with arguments that I can't really understand:
**- "I'd say that's not a reason enough to change something that has little practical benefit, it's not critical"**
So only the only UX changes that will be considered are critical ones? I don't think that is a good point of view to improve UX
**- "Well I don't see how it's any better to confuse people with a new name and create inconsistencies with third party content etc. than to have a cumbersome but well established name"**
That's a classic "we always did it this way" approach which I don't think Godot is in the position in when it comes to the Editor UX/UI.
I remember very vividly how the Blender community raged when it was changed to default left-click. But the software benefitted hugely from it.
Also things are changed all the time making third party content inconsitent. Most of the comments on my YouTube tutorials (which are mostly for 3.x) are asking about breaking changes. But people manage to understand that and those are much more complicated than just renaming something that is obviously the same thing.
**- "If they're critical enough in breaking change we don't do them until the next major version, if we do them at all, when we can't guarantee stability or effectiveness of that change"**
This change would not be critical, but it's also (from my pov) not a **breaking** change. A breaking change is a change that requires the user to change their project to make it work again. It breaks components or makes something that worked before not work anymore. 

Is changing a label a breaking change?
It would make some outside references not reflect the current state anymore - but I would argue that this is true for 90% of the content out there. Like I said before, things change all the time.

So overall my impression from this discussion was that there were a lot of reasons gives why not to do it and very few answers *how to move forward* with it - that's why I commented.

**Reply by AThousandShips** [reply]
Well as you can see I gave suggestions, and presented *my opinion* on why this change might cause negative consequences, and why it's in *my opinion* a change with minor benefits, and as I was asked about compatibility i explained it in broader terms, please read my messages again where I clearly say I'm not necessarily against this, or think it's an impossible change, and where I suggest ways to do so that's less disruptive :)

**Reply by andyprice** [reply]
Regarding "how to move forward" - I think spreading the word to get more upvotes on the proposal is a good next step :)

**Comment by passivestar** [comment]
Just want to express my full support for this change, I explained my reasoning [here](https://github.com/godotengine/godot-proposals/discussions/9071), and I'm really looking forward for this to happen at whatever dev cycle it will be most convenient

**Comment by TheDuriel** [comment]
Definitely in favor of the squash in character length.

Though I'd note that a **more correct** label would be "Resource(s)" since Godot does not work with files, but with resources. The makeup of the contents of the File dock happen to represent the underlying file system, mostly, but **is not** that.

**Reply by passivestar** [reply]
Since that particular dock allows you to create resources only on disk (as opposed to for example the Inspector dock that allows you to create resources in-memory), doesn't it make sense for it to be called Files as it's more specific?

**Reply by TheDuriel** [reply]
File is better than File System.

Both are equally incorrect.

**Reply by AThousandShips** [reply]
Also the dock shows things like assets (not resources, but are imported) and non-resource files (such as `.txt`), the fact that it doesn't show all files doesn't make the name `Files` or `File System` wrong though, otherwise the default settings would make the file explorer in Windows not correct either (as it hides hidden files)

**Reply by TheDuriel** [reply]
The dock exclusively shows resource files. All files it displays are imported and openable **as resources** by Godot. That includes the recently added support for .txt files. GLTF files **are** resources which are additionally broken out into a scene resource and mesh/etc subresources.

TLDR: Those are resources, you just did not notice.

The Dock does in fact not show **any** non resource files. It is **incapable** of doing so. Since to show a file, it must be registered as an importable resource.

**Reply by AThousandShips** [reply]
No, `gltf` files are not resource files, nor are `.txt` files, try `load("res://my_text.txt")` and you'll see (also `.txt` files are not exported, even when selecting "All Resources", unlike resources)

So I haven't failed to notice anything 🙃

**Reply by TheDuriel** [reply]
You are confusing the ability to load a **file path** and get a resource object, with godot treating them as a resource.

Loading a GLTF works fine in fact, because they are resources just how png files are resources. They are registered as a resource file type in the resource management system, and associated with an resource importer and a resource loader.

.md, .txt, .json files, are also registered as resource in the resource manager, **or they would not appear in the file dock**, and all of them have a resource importer. Only json however implements a resource loader.

**Reply by TheDuriel** [reply]
https://github.com/godotengine/godot/blob/master/scene/resources/text_file.h

Text File is literally a resource type. And this resource is used to let you see and edit them.

**Reply by AThousandShips** [reply]
No, `txt` files do not have importers... But what ever, you can look it up yourself if you think I'm wrong, I'm done

> Text File is literally a resource type. And this resource is used to let you see and edit them.

No it's not, it's a `Resource` class, that's different. It's not exposed and doesn't have a loader, it's not a `txt` file...

Have a nice day

**Reply by timothyqiu** [reply]
Technically, only resources are shown in the FileSystem dock.

`gltf` files are shown because they are recognized by an importer and converted into resources internally.

`txt` files are shown simply because it's a hack. There are indeed no importers, but a list of extensions are kept as exceptions. Matching files are considered virtual `TextFile` resources.

**Reply by tomsterBG** [reply]
I was almost going to agree, but then remembered, the File system is just a convenient, in-editor way to represent only the files that are necessary for the engine's editor. I often use .md files which are different that .txt. However i have not seen a resource called Folder that works like a folder. Given these things, i believe a custom dock someone would make in their own engine fork will deserve the name "Resources" more than this engine-specific representation of what's on your disk.

Imo, it should be "Files".

**Comment by ibito** [comment]
Even though I like the shorter name more, I think "FileSystem" is more accurate, because it's a file system inside your project, hence the "res://" and "user://" I believe.

So, if this tab ends up saying "Files" instead, I'd like it.

**Comment by jmarceno** [comment]
Files, is wrong and misleading as we are hiding files from the user and not giving it an easy way to see it. (.import and anything that is not "recognised" by the engine). File managers can hide files, but you can normally revert that with a single checkbox. Not the case here.

I like the short name. Not against, but it is wrong imo.
I would prefer Content. But I do not feel that strong about this label.

**Reply by LauraWebdev** [reply]
By that logic (that a "Files" tab is misleading since it's hinding files), "FileSystem" is also wrong.

**Reply by jmarceno** [reply]
Yes. That is why I said I prefer Content.
But I really think all this discussion over a label is very funny 🙂

**Reply by LauraWebdev** [reply]
@jmarceno I just edited my top level comment, I do prefer Content/Ressources over Files/FileSystem, since it is NOT a FileSystem/File list.

**Reply by ibito** [reply]
From the top of my head, what about "Assets"? Since everything in there is assets... unless it is not haha

**Comment by LauraWebdev** [comment]
I do think "Files" is the way to go, we don't call the Scenes tab "SceneTree" either.
Edit: After thinking about it for a few minutes, I do like "Ressources", as it more clearly conveys the information that it is not just the filesystem.

**Comment by DaisyGames** [comment]
I'd like to voice my support for adjusting the name. I think Files is a very good option, if some find it too vague Project Files would be another good fit in my opinion.

While I am sure this is a very trivial proposal, I believe this should be addressed. FileSystem (There isn't even a space between the words.) is something I'd expect to see in an alpha/beta of a game engine, not a stable release, especially in Godot 4.

Essentially why I'm in favor is that appearances and polish matter. I find FileSystem to be as strange as having a typo somewhere in the editor UI. Imagine if instead of Inspector it would say Inspctor. Does it change how Godot works? No. Does it make Godot look worse? Yes.

**Comment by ibito** [comment]
What about "Assets"? Everything in there is just assets isn't it? (Could be wrong in this assumption).

**Reply by DaisyGames** [reply]
Since this also shows script files, I don't think Assets would work.

**Reply by ibito** [reply]
I always thought scripts were also assets, oopsie! (Not a pro game dev as you see).

**Reply by tomsterBG** [reply]
I agreed with assets until i realized, in a normal project your pattern is to name folders addons, scripts, scenes, assets, resources, etc. So naming the tab "Assets" implies that it is specifically for assets, but this Files tab is more abstract than that so i think it should be named Files.

**Comment by emanvidmaker** [comment]
I propose a happy medium; lets name it '_**Project**_` because it shows the project folder. Generic, simple, precise.

**Reply by elenakrittik** [reply]
Nice idea overall, but given we already have a "Project" menu at the top, "FileSystem" -> "Project" will cause a great deal of confusion even for long-time users, not to mention beginners and the ambiguity introduced in day-to-day speech when referring to "Project" (which one? you'll almost always have to explicitly specify).

**Comment by yosoyfreeman** [comment]
I personally don't believe any of the proposed names describes better what it is that the current one.

**Reply by LauraWebdev** [reply]
I do think "FileSystem" contributes to the initial confusion why certain file types are missing even if they ARE in the OS filesystem.

**Comment by muno777** [comment]
I'm in favor of Files. It fits in with the other tab names (meanwhile FileSystem is, unusually, pascalcase) and describes the function of the tab.

Also, the search bar already says "Filter Files"... 🤔

![image](https://github.com/godotengine/godot-proposals/assets/37985497/f66acff3-8e0f-448f-b9e7-e7f4c0827bcc)

**Comment by dmaz** [comment]
the fact that I can add ard remove files through the base OS file explorer makes me happy with 'FileSystem', 'Files' or even 'Project Files' I'm not a fan of "resources" unless godot actually managed this structure

**Comment by peterhoglund** [comment]
Unreal - Content Drawer or Content Browser
Unity - Project (and files are in the Asset folder)
Game Maker - Asset Browser
Construct - Project
CryEngine - Asset Browser
O3DE/Lumberyard - Asset Browser
Cocos2D - Assets

None of the other larger engines calls this Files or FileSystem or anything with files. They seem to agree on Content, Assets, or Project with a strong preference for Assets.

**Comment by ibito** [comment]
Since it has a `res://` what about "Resources"?

**Reply by Calinou** [reply]
This was suggested above: https://github.com/godotengine/godot-proposals/discussions/9067#discussioncomment-8443085

**Reply by LauraWebdev** [reply]
I do like this the most, as it mentally connects this tab with the res protocol and also is a textual indicator that this tab does NOT show all files from the OS.

**Comment by azur-wolve** [comment]
also its shorter so the tab takes less space when merged with other docks

**Reply by tomsterBG** [reply]
Exactly! I have to sometimes widen the tab to disable the sideways scrolling and arrows because this works fine for wide rows like above the scene viewport, but not on narrow rows like on every side dock.
<img width="195" height="90" alt="image" src="https://github.com/user-attachments/assets/bda1b495-3c53-4c94-88fa-34efdf3a0fdb" />
<img width="207" height="76" alt="image" src="https://github.com/user-attachments/assets/b0e47a4d-0d18-473b-a9e1-1f1a61605ab3" />

It's a small thing, but i believe people will benefit from it more than they will ever be "confused" or "mislead by an inaccurate name".

**Reply by andyprice** [reply]
One option for saving space is to change the `Interface->Editor->Dock Tab Style` setting to `Icon Only` (It's in Editor->Editor Settings...)

**Comment by scotmcp** [comment]
It makes sense, because I don't know how many times I have to distinguish between the Filesystem and the Filesystem...

"Go to your filesystem....no not there I mean in a Godot, yeah bottom left"  a few minutes later, "Ok no go to your filesystem....ugh, no this time I mean on your computer, yes that's right now drag that into the Filesystem....sigh, the one in your Godot engine...yes."

**Comment by tomsterBG** [comment]
I have seen some curious alternatives to "Files", such as "Assets" and "Resources". However if you see some tutorials or download projects or even look at your own project, there definitely is at least one asset folder and a resource folder and a script folder and a scene folder. It would cause a level of ambiguity and confusion to de-abstractify the name from Files to something more concrete like Assets and Resources.

I'm in my own project and want to use the Assets dock for my scripts, wait scripts aren't assets! Art and sounds are!
Let's use the Resources dock to find my .md files... wait! Are .md files resources? Technically they can be considered, yeah.
Let's use the Resources dock to find my scripts folder... wait, are folders resources? Are they?

So all in all the way it is in Godot specifically, I believe the best name is Files. It is as abstract as it needs to be and people who modify the engine can freely make their own docks named Resources and Assets without causing ambiguity.

As for Project, there is already a Project tab on the top and it wouldn't be cool for those to double down.

There's already ambiguity with the Scene dock and the Scene tab:
<img width="87" height="96" alt="image" src="https://github.com/user-attachments/assets/d8a0b173-0f27-426a-84aa-5658985d0b7a" />

Another reason for Files, it's the shortest good name proposed so far and specifically for Godot's tab system, it's best to keep names short in tall windows because they get cut off like this:
<img width="195" height="90" alt="image" src="https://github.com/user-attachments/assets/b290fd65-6f47-4885-9282-72c6a0b73497" />
<img width="207" height="76" alt="image" src="https://github.com/user-attachments/assets/c1f93072-61a3-4003-b0e0-44b9c8069a0e" />

All in all, many people say many things and i believe there are way more than enough logical arguments to support the "Files" name instead of any other name. It's the least confusing, the least ambiguous, the `Filter Files`  search will have to say `Filter Project` which... `Filter Assets` ... how's that sound? ... `Filter Resources` ... it just makes no sense:
<img width="196" height="162" alt="image" src="https://github.com/user-attachments/assets/2b2866e4-2980-4d79-a32a-eb26c07b9f60" />

Godot is different than other engines and it just works the best this way imo.

**Reply by Miku0Sama** [reply]
> However if you see some tutorials or download projects or even look at your own project, there definitely is at least one asset folder and a resource folder and a script folder and a scene folder.

You guys are using folders? You mean you don't just dump everything in one folder with zero organization?

Ok but seriously, "Files" is probably the best name for it. A reasonable user can assume that "Files" is simply referring to all files that are relevant to the Godot engine and not "Every single file on you system within the project folder". But "FileSystem" definitely has a stronger implication of that, first time I opened up a project folder in my windows file explorer I was shocked to see that there was a large number of things there that were not displayed within the editor's UI.

The only other option I'd argue is equally good would be "Resources" but that's barely any smaller in width than "FileSystem" so I'm obviously gonna cast my vote for the more compact choice.
